subsys/net/lib/sockets/sockets.c

```diff
@@ -1134,6 +1134,54 @@ static int sock_get_pkt_src_addr(struct net_pkt *pkt,
 	return ret;
 }
 
+#if defined(CONFIG_NET_OFFLOAD)
+static int sock_get_offload_pkt_src_addr(struct net_pkt *pkt,
+					 struct sockaddr *addr,
+					 socklen_t addrlen)
+{
+	int ret = 0;
+
+	if (!addr || !pkt) {
+		return -EINVAL;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_IPV4) &&
+	    net_pkt_family(pkt) == AF_INET) {
+		if (addrlen < sizeof(struct sockaddr_in)) {
+			ret = -EINVAL;
+			goto error;
+		}
+
+		memcpy(addr, &pkt->remote,
+		       MIN(addrlen, sizeof(pkt->remote)));
+	} else {
+		ret = -ENOTSUP;
+		goto error;
+	}
+
+error:
+
+	return ret;
+}
+
+static bool net_pkt_addr_is_unspecified(struct net_pkt *pkt)
+{
+	return net_ipv4_is_addr_unspecified((struct in_addr *) &pkt->remote);
+}
+#else
+static int sock_get_offload_pkt_src_addr(struct net_pkt *pkt,
+					 struct sockaddr *addr,
+					 socklen_t addrlen)
+{
+	return 0;
```

Comment:
To avoid complaints from compiler (with certain options), add here
```
ARG_UNUSED(pkt);
ARG_UNUSED(addr);
ARG_UNUSED(addrlen);
```

Reply:
added